accepted/0000-remove-yarn-check.md

```diff
@@ -0,0 +1,27 @@
+- Start Date: 30 Oct 2018
+- RFC PR: https://github.com/yarnpkg/rfcs/pull/106
+- Yarn Issue: n/a
+
+# Remove `yarn check`
+
+## 1. Motivation
+
+Running `yarn install` should work out of the box. There's no reason for `yarn check` to exist from a user perspective - they should never have to run it, because installs must always be right.
+
+Being not particularly useful, the `yarn check` command receives less attention and as a result it often yields wrong / confusing results. This leads users to believe that `yarn install` is broken when it's in fact `yarn check` (https://twitter.com/betaorbust/status/1055610508533878784).
+
+## 2. Detailed Design
+
+We'll remove the `yarn check` command in Yarn 2.0.
```

Comment:
As well as `yarn install --check-files`, I assume?

Reply:
So `--check-files` is a bit different - it doesn't actually check 😅 

It has to do with the way we make an early bailout if the `node_modules` is already there (to avoid running an installation if there is no need). Without `--check-files` Yarn will only checks for the various folders to exist (plus various other settings), whereas with `--check-files` it will also check for all the registered files to exist. It's not super useful anymore, but it's not as redundant as `yarn check`.

Reply:
Oh, I see. That’s unfortunate, I like the integrity check of `—check-files`, but for most people I can see checking of the expected directories exist is probably enough.